src/averageAges.js

```diff
@@ -54,8 +68,53 @@ function calculateWomenAverageAge(people, withChildren) {
  *
  * @return {number}
  */
-function calculateAverageAgeDiff(people, onlyWithSon) {
-  // write code here
+function calculateAverageAgeDiff(people, onlyWithSon = false) {
+  let children = people.filter(person => person.mother !== null);
+
+  if (onlyWithSon) {
+    children = filterPersonSex(children, 'm');
+  }
+
+  const womenOnly = filterPersonSex(people, 'f');
```

Comment:
There is no need to use word `Only`
Also, there is no need to filtrate by sex and after make a second filtration by name